servicetalk-concurrent-internal/src/testFixtures/java/io/servicetalk/concurrent/internal/TimeoutTracingInfoExtension.java

```diff
@@ -18,23 +18,86 @@
 
 import org.junit.jupiter.api.extension.AfterEachCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.lang.invoke.MethodHandles;
 import java.lang.management.LockInfo;
 import java.lang.management.ManagementFactory;
 import java.lang.management.MonitorInfo;
 import java.lang.management.ThreadInfo;
 import java.lang.management.ThreadMXBean;
+import java.util.Collections;
 import java.util.Comparator;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-public class TimeoutTracingInfoExtension implements AfterEachCallback {
+import static java.util.concurrent.TimeUnit.DAYS;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.regex.Pattern.CASE_INSENSITIVE;
+import static java.util.regex.Pattern.UNICODE_CASE;
+
+/**
+ * Junit extension which will print information about all threads if unit test method throws
+ * {@link TimeoutException}.
+ */
+public final class TimeoutTracingInfoExtension implements AfterEachCallback {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
Interesting approach for static class lookup that prevents incorrect class name if the logger is copy-pasted, TIL :)
I use Intellij IDEA Inspection to catch cases when the logger doesn't match the class name.

Reply:
that's copy/paste proof method. Plus I have an IDEA shortcut  - "log" + <TAB> got expanded into this line above